### ISSUE TITLE
doesn't conform to the bound 'StateStreamable<state>' of the type parameter 'B'.

### ISSUE THREAD

**Issue by serkanagca34** [felangel/bloc#3322]
why am i getting the error? I couldn't find the reason. I'll be happy if you can help me. @felangel 

![Untitled](https://user-images.githubusercontent.com/62822922/162732990-35bad557-7fdf-4fe6-aec3-e245f6f8159a.png)

Cubit File:
`import 'package:bitistanbulmobile/models/home_market/market_pair_data.dart';
import 'package:bitistanbulmobile/services/rest_api/rest_data_source.dart';
import 'package:bloc/bloc.dart';
import 'package:equatable/equatable.dart';

part 'buy_sale_state.dart';

class BuySaleCubit extends Cubit<BuySaleState> {
  BuySaleCubit() : super(BuySaleInitial());
  final RestDataSource api = RestDataSource();
  Future<void> getPairData() async {
    emit(BuySaleLoading());
    try {
      final response = await api.fetchMarketPair();
      final List<MarketPairResult>? pairData = response.result;
      emit(BuySaleLoaded(response: pairData ?? []));
    } catch (e) {
      emit(BuySaleError(e.toString()));
    }
  }
}
`

State File:
`part of 'buy_sale_cubit.dart';

abstract class BuySaleState extends Equatable {
  const BuySaleState();

  @override
  List<Object> get props => [];
}

class BuySaleInitial extends BuySaleState {
  const BuySaleInitial();
}

class BuySaleLoading extends BuySaleState {
  const BuySaleLoading();
}

class BuySaleLoaded extends BuySaleState {
  final List<MarketPairResult> response;
  BuySaleLoaded({required this.response});
}

class BuySaleError extends BuySaleState {
  final String errorMessage;
  const BuySaleError(this.errorMessage);
}
`

**Comment by marcossevilla** [comment]
hi @serkanagca34 👋  you should specify the type of the state in your cubit, like this:
```dart
class BuySaleCubit extends Cubit<BuySaleState> {
  ...
}
```

**Comment by serkanagca34** [comment]
> merhaba @serkanagca34 👋 kübitinizde durumun türünü şu şekilde belirtmelisiniz:
> 
> ```dart
> class  BuySaleCubit , Cubit'i  genişletiyor  < BuySaleState > {
>   ...
> }
> ```

I did as you said, unfortunately it didn't work.

**Comment by felangel** [comment]
@serkanagca34 can you share a link to a complete, minimal reproduction sample? Thanks!

**Comment by serkanagca34** [comment]
hello i solved the problem. The problem was giving an error because there were 2 classes with the same name, so my StatefulWidget class name was the same as my block state class name :) It was fixed when I changed the name of my block state class.  thank you for your interest. @felangel 

![Screenshot 2022-04-13 132936](https://user-images.githubusercontent.com/62822922/163161311-9c7b9022-f773-4794-a122-344802cad83a.png)


![Screenshot 2022-04-13 133630](https://user-images.githubusercontent.com/62822922/163161029-4d8cb351-f469-4911-aae4-c45636208ecc.png)
